python/ray/client_builder.py

```diff
@@ -0,0 +1,99 @@
+import os
+import importlib
+import logging
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Tuple
+
+from ray.ray_constants import RAY_ADDRESS_ENVIRONMENT_VARIABLE
+from ray.job_config import JobConfig
+import ray.util
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ClientInfo:
+    """
+    Basic information of the remote server for a given Ray Client connection.
+    """
+    dashboard_url: str
+    python_version: str
+    ray_version: str
+    ray_commit: str
+    protocol_version: str
+
+
+class ClientBuilder:
+    """
+    Builder for a Ray Client connection.
+    """
+
+    def __init__(self, address: Optional[str]) -> None:
+        self.address = address
+        self.job_config = JobConfig()
+
+    def env(self, env: Dict[str, Any]) -> "ClientBuilder":
+        """
+        Set an environment for the session.
+        """
+        self.job_config = JobConfig(runtime_env=env)
+        return self
+
+    def connect(self) -> ClientInfo:
+        """
+        Begin a connection to the address passed in via ray.client(...).
+        """
+        client_info_tuple = ray.util.connect(
+            self.address, job_config=self.job_config)
+        return ClientInfo(
+            dashboard_url=None,
+            python_version=client_info_tuple["python_version"],
+            ray_version=client_info_tuple["ray_version"],
+            ray_commit=client_info_tuple["ray_commit"],
+            protocol_version=client_info_tuple["protocol_version"])
+
+
+class _LocalClientBuilder(ClientBuilder):
+    pass
+
+
+def _split_address(address: str) -> Tuple[str, str]:
+    """
+    Splits address by "://" into a module string and an inner_address.
+    NOTE: inner_address may contain "://"
+    """
+    module_start_idx = address.find("://")
```

Comment:
if :// does not exist, this returns -1, which results in a wrong behavior. Can we raise an exception of :// is not found?

Reply:
I can raise an exception. The case where `find(://) == -1` should be handled before this function is invoked.

Reply:
@ericl Sure! That is definitely the correct way to handle this :)